euth/communitydebate/views.py

```diff
@@ -29,15 +32,28 @@ def get_queryset(self):
 class TopicDetailView(generic.DetailView):
     model = communitydebate_models.Topic
     queryset = \
-        communitydebate_models.Topic.objects.annotate_positive_rating_count() \
+        communitydebate_models.Topic.objects\
+        .annotate_positive_rating_count() \
         .annotate_negative_rating_count()
     permission_required = 'euth_communitydebate.view_topic'
 
+    def get_context_data(self, **kwargs):
+        context = super().get_context_data()
+        upload_files = communitydebate_models.TopicFileUpload.objects\
+            .filter(topic=self.object)
+        context['upload_files'] = upload_files
+        return context
+
 
-class TopicCreateView(PermissionRequiredMixin, generic.CreateView):
+class TopicCreateView(PermissionRequiredMixin, generic.CreateView,
+                      TopicFormMixin):
     model = communitydebate_models.Topic
     form_class = forms.TopicForm
     permission_required = 'euth_communitydebate.propose_topic'
+    template_name = 'euth_communitydebate/topic_form.html'
+    # hack to make rendering of error messages work when topic
+    # has not been saved yet
+    object = None
```

Comment:
It's nearly right! If you just move it to the post method it's the same way as it's handled in django and all is fine (and still working)!
https://github.com/django/django/blob/d36413a2eb5678dda35559a48769703acd784898/django/views/generic/edit.py#L170